### PR TITLE
[CI] enable new runners for some of jobs

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
in some of the jobs, we enable intel-maintained, ephemeral runners.

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

File: .github/workflows/bandit.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
```
Comment (lplewa) [inline .github/workflows/bandit.yml:21]: not needed anymore - also we lots windows build here - not sure if is by desing

File: .github/workflows/proxy_lib.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
         build_type: [Release, Debug]
         compiler: [{c: gcc, cxx: g++}]
         proxy_lib_pool: ['SCALABLE', 'JEMALLOC']
-    runs-on: ${{matrix.os}}
+    runs-on: ${{ github.repository_owner == 'oneapi-src' && 'intel-ubuntu-22.04' || 'ubuntu-latest' }}
```
Comment (lplewa) [inline .github/workflows/proxy_lib.yml:23]: update matrix

**Review by lplewa — CHANGES_REQUESTED** [review]

File: .github/workflows/bandit.yml (hunk 1 above)
Comment (lukaszstolarczuk) [reply .github/workflows/bandit.yml:21]: Needed, needed. Fixed.

File: .github/workflows/proxy_lib.yml (hunk 2 above)
Comment (lukaszstolarczuk) [reply .github/workflows/proxy_lib.yml]: done.

**Review by pbalcer — APPROVED** [review]

**Review by lplewa — APPROVED** [review]

**Pull request merged**